src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DefaultRazorProjectFileSystem.cs

```diff
@@ -23,67 +40,93 @@ public override IEnumerable<RazorProjectItem> EnumerateItems(string basePath)
     {
         var absoluteBasePath = NormalizeAndEnsureValidPath(basePath);
 
-        var directory = new DirectoryInfo(absoluteBasePath);
```

Comment:
I would think that using `DirectoryInfo` has some advantages too, e.g., the DirectoryInfo holds on to the directory handle so it doesn't have to be searched for it as is the case when using the static methods which take string paths (which could result in less syscalls). Not sure if that's better or worse than the allocations.